rfcs/reftest_simplification.md

```diff
@@ -0,0 +1,56 @@
+# RFC #15: Reftest graph simplification
+
+## Summary
+
+Simplify what we allow within the reftest graph.
+
+## Details
+
+### Background
+
+Currently, reftests form a graph which can quickly end up with a large number of pass conditions.
+
+In rough principle, we load everything with `<link rel=match>`/`<link rel=mismatch>` into a directed graph (with URLs as vertices, `match`/`mismatch` as edges), and then make every source node (i.e., a node with no incoming edges) into a "test".
+
+From each test, we enumerate all possible walks from the source node with either no repeating nodes and ending at a sink node (i.e., a node with no outgoing edges) or a ending at the first repeated node.
+
+In reality, we have very little complexity within the graph, as of [37e794f690](https://github.com/web-platform-tests/wpt/tree/37e794f69091ffed1dfa7399a3ef154125fddda9): we have 14954 tests, and only 219 are more complex than a simple two page comparison.
+
+Of those 219:
+ * 111 just have alternates at the top level (the basic OR case),
+ * 55 are trails with no repeating vertices (the basic AND case),
+ * 48 are trails ending in a cycle (still a basic AND),
```

Comment:
IIRC, the docs used to recommend things with multiple refs form cycles because it means if you link as `==` to one ref you get `==` to all of them and it doesn't matter which you link to first.

Reply:
Makes sense, thanks.